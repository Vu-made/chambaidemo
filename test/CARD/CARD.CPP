#include <bits/stdc++.h>
#define N 1000005
using namespace std;
int Next[N], Pre[N];
int n, k;
string s;
int main()
{
    freopen("CARD.INP","r",stdin);
    freopen("CARD.OUT","w",stdout);
    cin >> n >> k >> s;
    Next[0] = 1; Pre[n+1] = n;
    for (int i=1; i<=n; i++)
    {
        Next[i] = i+1;
        Pre[i] = i-1;
    }
    int p;
    for (int i=0; i<s.size()-1;i++)
    {
        if (s[i] == 'A')
        {
            p = Next[0];
            Next[0] = Next[p];
            Pre[Next[p]] = 0;
        }
        else
        {
            p = Next[Next[0]];
            Next[Pre[p]] = Next[p];
            Pre[Next[p]] = Pre[p];
        }
        Next[p] = n + 1;
        Pre[p] = Pre[n+1];
        Next[Pre[n+1]] = p;
        Pre[n+1] = p;
    }
    p = Next[0];
    for (int i=1; i<=k; i++) p = Next[p];
    cout << Pre[p]-1 << " " << p-1 << " " <<Next[p]-1;
    return 0;
}
