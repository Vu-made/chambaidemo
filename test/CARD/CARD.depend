# depslib dependency file v1.0
1602669964 source:c:\users\admin\desktop\bgdv\bt\thuat_toan\array1d_3\card\card.cpp
	<bits/stdc++.h>

