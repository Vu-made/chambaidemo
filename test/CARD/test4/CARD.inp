1231 1229 BBBBBBBAABBBBBAABBBBBAABBBBAABBBBBAABBBBBAABBBBBAABABABABABABABABABA.
