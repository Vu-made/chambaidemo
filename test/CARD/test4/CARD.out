64 67 66
